packages/amplify-provider-awscloudformation/lib/update-idp-roles-cfn.json

```diff
@@ -0,0 +1,173 @@
+{
+
+    "UpdateRolesWithIDPFunction": {
+        "DependsOn": [
+            "AuthRole",
+            "UnauthRole"
+        ],
+        "Type": "AWS::Lambda::Function",
+        "Properties": {
+            "Code": {
+                "ZipFile": {
+                    "Fn::Join": [
+                        "\n",
+                        [
+                            "const response = require('cfn-response');",
+                            "const aws = require('aws-sdk');",
+                            "let responseData = {};",
+                            "exports.handler = function(event, context) {",
+                            "  try {",
+                            "    let authRoleName = event.ResourceProperties.authRoleName;",
+                            "    let unauthRoleName = event.ResourceProperties.unauthRoleName;",
+                            "    let idpId = event.ResourceProperties.idpId;",
```

Comment:
Just for clarification, `event.ResourceProperties.idpId` is the current Identity Pool that will give us the following policy:

```
      "Condition": {
        "StringEquals": {
          "cognito-identity.amazonaws.com:aud": "us-east-1:c04c1499-52cf-4df7-97ca-0bf60eecfdd8"
        }
    }
```